docs/spfx/web-parts/guidance/getting-started-with-top-actions.md

```diff
@@ -0,0 +1,151 @@
+
+```typescript
+return {
+  ...
+  onExecute: (actionName: string, newValue: any) => {
+    if (actionName === 'reset') {
+      // user defined logic to reset the web part
+      this.reset();
+    }
+  }
+}
+```
+> Common pitfalls when implementing the `onExecute` command, is not syncing the new state with the web part properties and/or not refreshing or re-rendering the web part.
+
+## Code Snippets
+### Button command
+The type interace for a button is similar to the property panel's button (`IPropertyPaneButtonProps`).
+
+```typescript
+import { ITopActions } from '@microsoft/sp-top-actions';
+import { PropertyPaneFieldType } from '@microsoft/sp-property-pane';
+...
+public getTopActionsConfiguration(): ITopActions | undefined {
+  return {
+    topActions: [
+      {
+        targetProperty: 'reset',
+        type: PropertyPaneFieldType.Button,
+        properties: {
+          icon: 'Reset'
+        }
+      }
+    ],
+    onExecute: (actionName: string, newValue: any) => {
+      if (actionName === 'reset') {
+        // user defined logic to reset the web part
+        this.reset();
+      }
+    }
+  };
+}
+```
+
+### Drop down command
+The type interace for a drop down is similar to the property panel's choice group (`IPropertyPaneChoiceGroupOption`).
+
+```typescript
+import { ITopActions } from '@microsoft/sp-top-actions';
+import { PropertyPaneFieldType } from '@microsoft/sp-property-pane';
+...
+public getTopActionsConfiguration(): ITopActions | undefined {
+  return {
+    topActions: [{
+      targetProperty: 'layout',
+      type: PropertyPaneFieldType.ChoiceGroup,
+      properties: {
+        options: [
+          {
+            // key maps to newValue in onExecute
+            key: 'card',
+            text: 'Card Layout',
+            imageSize: { width: 32, height: 32 },
+            iconProps: { officeFabricIconFontName: 'ArticlesIcon' },
+            checked: this.state.layout === 'card'
+          },
+          {
+            key: 'list',
+            text: 'List Layout',
+            imageSize: { width: 32, height: 32 },
+            // you can use iconProps, icon to define icons
+            icon: 'List',
+            checked: this.state.alignment === 'list'
+          }
+        ]
+      }
+    }],
+    // for ChoiceGroup drop down, the newValue tells us which option's key was selected
+    onExecute: (actionName: string, newValue: any) => {
+      if (actionName === 'layout') {
+        this.setLayout(newValue);
+        this.render();
+      }
+    }
+  };
+}
+```
+
+## Advanced configurations
+For advanced configurations of your top action commands, checkout the type definitions from `@microsoft/sp-property-pane`. Currently, the two supported topActions, button and drop down, can be defined using the same structure as `IPropertyPaneChoiceGroupOption` and `IPropertyPaneButtonProps`. From here you can see how to use features like icons, aria labels, disable state, and more.

```

Comment:
Can you please describe what properties from those interfaces are actually working? I believe not all of them are in use and will be added to actual interfaces in the next iteration.